server/src/com/thoughtworks/go/server/security/RemoveAdminPermissionFilter.java

```diff
@@ -89,4 +94,9 @@ public void onConfigChange(CruiseConfig newCruiseConfig) {
     private SecurityConfig securityConfig(CruiseConfig newCruiseConfig) {
         return newCruiseConfig.server().security();
     }
+
+    @Override
+    public void onAuthorizationPluginUnload() {
```

Comment:
This callback happens in case of plugin unload as well as when plugin initiates a request to invalidate the roles cache. May be rename the interface and method to suit both workflows